Refactor series to new create node data pattern pt.4

alantreadway opened the pull request:
https://ag-grid.atlassian.net/browse/AG-16475

alantreadway committed:
AG-16475 - Factor out common createNodeData() patterns.

alantreadway committed:
AG-16475 - Factor out common createNodeData() patterns.

alantreadway committed:
AG-16475 - Factor out common createNodeData() patterns.

alantreadway committed:
Fix setupworktree.sh.

alantreadway committed:
AG-16475 - Code review cleanup for createNodeData() refactoring.

- Remove unused utility functions from cartesianSeriesUtil.ts
- Move Waterfall pointData from instance field to context object
- Rename RangeArea existingMarkerData to existingNodes for consistency
- Simplify Heatmap labels initialization (always rebuilt, not incremental)

lsjroberts commented on packages/ag-charts-community/src/chart/series/cartesian/cartesianSeriesTypes.ts:
````diff
@@ -33,6 +34,100 @@ export interface CartesianSeriesNodeDataContext<
+ * }
+ * ```
+ */
+export interface CartesianCreateNodeDataContextBase<TDatum extends CartesianSeriesNodeDatum> {
````
It doesn't look like anything here extends this except `CartesianCreateNodeDataContext`?

alantreadway replied:
Good spot, removed.

lsjroberts commented on packages/ag-charts-community/src/chart/series/cartesian/cartesianSeries.ts:
```diff
@@ -409,6 +411,258 @@ export abstract class CartesianSeries<TTypes extends CartesianSeriesTypes> exten
+     * @param existingNodes - The existing node array from previous createNodeData()
+     * @param additionalCondition - Optional series-specific condition for incremental updates
+     */
+    protected createIncrementalUpdateState<T>(
```
Is this function used?

alantreadway replied:
Good spot, removed.

lsjroberts commented on packages/ag-charts-enterprise/src/series/waterfall/waterfallSeries.ts:
```diff
@@ -429,22 +430,24 @@ export class WaterfallSeries extends _ModuleSupport.AbstractBarSeries<WaterfallS
+            animationEnabled,
             xKey,
-            yKey,
+            yKey: yKey,
```
nit

alantreadway replied:
Fixed

lsjroberts reviewed (approved):
LGTM

alantreadway committed:
AG-16475 - Address PR feedback: remove dead code and fix style nit.

- Remove unused createIncrementalUpdateState() helper function
- Fix property shorthand inconsistency (yKey: yKey -> yKey)

alantreadway committed:
AG-16475 - Remove redundant interface.
